### PR TITLE
chore(quantic): prepare quantic workflows to use new environment variables / secrets

### PR DESCRIPTION
**Pull request by fbeaudoincoveo**
https://coveord.atlassian.net/browse/KIT-3071

### CODE REVIEW

File: .github/actions/e2e-quantic-setup/action.yml [hunk 1]
```diff
@@ -13,9 +13,9 @@ runs:
     - run: npx nx run quantic:build
       shell: bash
     - run: |
-        echo "${{ inputs.jwtkey }}" > ${{ env.SFDX_AUTH_JWT_KEY_FILE }}
+        echo "${{ inputs.jwtkey }}" > server.key
         npx --no-install ts-node scripts/build/deploy-community.ts --ci
-        rm ${{ env.SFDX_AUTH_JWT_KEY_FILE }}
+        rm server.key
```
Comment (fbeaudoincoveo) [inline .github/actions/e2e-quantic-setup/action.yml:18]: I set this as an environment variable in Quantic Staging / Production, but for e2e we can just use a constant. It's not sensitive so it doesn't have to be a repo secret

File: .github/workflows/release.yml [hunk 2]
```diff
@@ -89,14 +89,14 @@ jobs:
   #     - uses: ./.github/actions/setup-sfdx
   #     - name: Promote SFDX package to production
   #       run: |
-  #         echo "${{ secrets.SFDX_AUTH_PACKAGE_JWT_KEY }}" > server.key
+  #         echo "${{ secrets.SFDX_AUTH_JWT_KEY }}" > ${{ vars.SFDX_AUTH_JWT_KEY_FILE }}
```
Comment (fbeaudoincoveo) [inline .github/workflows/release.yml:92]: The job is still commented out at this point, but I'm getting it ready

File: .github/workflows/create-quantic-package.yml [hunk 3]
```diff
@@ -45,26 +46,10 @@ jobs:
...
+          echo "${{ secrets.SFDX_AUTH_JWT_KEY }}" > ${{ vars.SFDX_AUTH_JWT_KEY_FILE }}
           npx --no-install ts-node scripts/build/create-package.ts --remove-translations --ci
-          rm server.key
+          rm ${{ vars.SFDX_AUTH_JWT_KEY_FILE }}
         env:
-          SFDX_AUTH_JWT_USERNAME: sfdc.integration.devv2.hub@coveo.com
-          SFDX_AUTH_JWT_KEY_FILE: server.key
-          SFDX_AUTH_CLIENT_ID: ${{ secrets.SFDX_AUTH_PACKAGE_CLIENT_ID }}
-
-  report-status:
```
Comment (fbeaudoincoveo) [inline .github/workflows/create-quantic-package.yml:56]: As discussed with @erocheleau, the GitHub Slack integration has made this job obsolete so we can now remove it.

TODO: delete the SLACK_WEBHOOK_CREATE_QUANTIC_PACKAGE repository secret

File: .github/actions/e2e-quantic-setup/action.yml (hunk 1 above)
Comment (fbeaudoincoveo) [reply .github/actions/e2e-quantic-setup/action.yml:18]: TODO: remove the `SFDX_AUTH_JWT_KEY_FILE` and `SFDX_AUTH_JWT_USERNAME` repo secrets

File: .github/workflows/release.yml (hunk 2 above)
Comment (louis-bompart) [reply .github/workflows/release.yml:92]: You're reserving this for KIT-3072 I guess?

**Comment by fbeaudoincoveo** [comment]
I documented the environments in the repo wiki:

https://github.com/coveo/ui-kit/wiki/Deployment-environments

File: .github/workflows/release.yml (hunk 2 above)
Comment (fbeaudoincoveo) [reply .github/workflows/release.yml:92]: Aye, although at this point it's probably going to be trivial

**Comment by mmitiche** [comment]
> I documented the environments in the repo wiki:
> 
> https://github.com/coveo/ui-kit/wiki/Deployment-environments
> 
> The environments themselves are here:
> 
> * [Quantic Staging](https://github.com/coveo/ui-kit/settings/environments/2646622374/edit)
> * [Quantic Production](https://github.com/coveo/ui-kit/settings/environments/2546058554/edit)

Nice, thank you 👍

**Review by mmitiche — APPROVED** [review]

**Review by louis-bompart — APPROVED** [review]
LGTM! Small changes in the vars of prod to match stg and gg!

**Comment by fbeaudoincoveo** [comment]
> LGTM! Small changes in the vars of prod to match stg and gg!

Fixed, thanks!

**Pull request merged**